rust/experimental/query_engine/engine-recordset/src/scalars/text_scalar_expressions.rs

```diff
@@ -51,6 +51,164 @@ where
                 || format!("Evaluated as: '{v}'"),
             );
 
+            Ok(v)
+        }
+        TextScalarExpression::Match(m) => {
+            // Execute the input expression to get the string to match
+            let input_value =
+                execute_scalar_expression(execution_context, m.get_input_expression())?;
+            let input_string = match input_value.to_value() {
+                Value::String(s) => s.get_value(),
+                _ => {
+                    return Err(ExpressionError::ValidationFailure(
+                        m.get_query_location().clone(),
+                        format!(
+                            "Text match input must be a string, got: '{:?}'",
+                            input_value.get_value_type()
+                        ),
+                    ));
+                }
+            };
+
+            // Execute the pattern expression to get the regex
+            let pattern_value = execute_scalar_expression(execution_context, m.get_pattern())?;
+            let pattern_string = match pattern_value.to_value() {
+                Value::String(s) => s.get_value(),
```

Comment:
User is forced to provide a string and then we compile a regex. Ouch! 🤮 Super bad for perf. What users should do is use parse to set the regex to a constant (`let r = parse('regex');`) so it is only compiled once. But for that to work what this code needs to do is check if the pattern is a regex. If so, use it. If the pattern is a string, it can fallback to compile mode. 

Should probably add a helper in `Value` for that. Could be `convert_to_regex` which returns the regex if the value is already one otherwise builds one up for strings. Similar pattern used already in the convert methods.